### ISSUE TITLE
Introduce mockito for mocking in tests

### ISSUE THREAD

**Issue by robintw** [debrief/pepys-import#860]
I recently came across the [mockito](https://mockito-python.readthedocs.io/en/latest/) library for creating mocks as part of tests. We use quite a lot of mocks, and the way we create them (using the built-in Python libraries for doing so) can be fairly confusing at times, and are potentially rather fragile. This library looks like a _far_ nicer way to create mocks, and so we should consider starting to use this - and potentially migrating some of our old tests to use this.

There is a good [walkthrough](https://mockito-python.readthedocs.io/en/latest/walk-through.html) on how the library works.

**Comment by IanMayo** [comment]
Yes, it looks good, and may reduce the volume of test code we need to create/maintain.
